opm/autodiff/RateConverter.hpp

```diff
@@ -450,7 +450,62 @@ namespace Opm {
                 }   
                 calcRmax();
             }
+
             /**
+             * Compute average hydrocarbon pressure and maximum
+             * dissolution and evaporation at average hydrocarbon
+             * pressure in all regions in field.
+             *
+             * Fluid properties are evaluated at average hydrocarbon
+             * pressure for purpose of conversion from surface rate to
+             * reservoir voidage rate.
+             *
+             * \param[in] state Dynamic reservoir state.
+             * \param[in] any  The information and communication utilities
+             *                 about/of the parallelization. in any parallel
+             *                 it wraps a ParallelISTLInformation. Parameter
+             *                 is optional.
+             */
+            template <typename ElementContext, class EbosSimulator>
+            void defineState(const EbosSimulator& simulator)
+            {
+
+                //const int numCells = cellPvtIdx_.size();
+                //const Region region = std::vector<int>(numCells, 0);
+                auto& ra = attr_.attributes(0);
+                auto& p  = ra.pressure;
+                auto& T  = ra.temperature;
+                std::size_t n = 0;
+
+                ElementContext elemCtx( simulator );
+                const auto& gridView = simulator.gridView();
+
+                const auto& elemEndIt = gridView.template end</*codim=*/0>();
+                for (auto elemIt = gridView.template begin</*codim=*/0>();
```

Comment:
I may be wrong, but does not this loop iterate over all cells present in the partition, and not just those *owned by* the partition? If so, the counts and averages will be slightly wrong, as overlap cells are considered multiple times. I believe a test like ` if (elem.partitionType() != Dune::InteriorEntity)` would resolve this?